### ISSUE TITLE
angdiff returns a numpy array when the function arguments are 2 scalars.

### ISSUE THREAD

**Issue by myeatman-bdai** [bdaiinstitute/spatialmath-python#81]
The docstring says this:
- ``angdiff(a, b)`` is the difference ``a - b`` wrapped to the range
      :math:`[-\pi, \pi)`.  This is the operator :math:`a \circleddash b` used
      in the RVC book
        - If ``a`` and ``b`` are both scalars, the result is scalar

The output of the function is this:
`In [9]: angdiff(10,0)
Out[9]: array([-2.56637061])`

**Comment by petercorke** [comment]
I've fixed `angdiff` and the unit tests.  Interestingly `self.assertEqual` from `unittest` and the equivalent from `numpy.testing` indicate that `1` is the same as `np.ndarray([1])` which is annoying.  I've added explicit tests on the return type. 

Now I have some kind of ssh key hell and I can't push to the repo, I think this is the first time I've tried since it moved to BDAII.  I have created a separate deploy key pair for this repo, since they can't be shared across users.  For now I'm out of ideas...
